src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,333 @@
+            var result = root.ReplaceNode(ifNode, invertedIf);
+            return document.WithSyntaxRoot(result);
+        }
+
+        private bool IsComparisonOfZeroAndSomethingNeverLessThanZero(BinaryExpressionSyntax binaryExpression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            var canSimplify = false;
+
+            if (binaryExpression.Kind() == SyntaxKind.GreaterThanExpression &&
+                binaryExpression.Right.Kind() == SyntaxKind.NumericLiteralExpression)
+            {
+                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
+                    binaryExpression.Left,
+                    (LiteralExpressionSyntax)binaryExpression.Right,
+                    semanticModel,
+                    cancellationToken);
+            }
+            else if (binaryExpression.Kind() == SyntaxKind.LessThanExpression &&
+                     binaryExpression.Left.Kind() == SyntaxKind.NumericLiteralExpression)
+            {
+                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
+                    binaryExpression.Right,
+                    (LiteralExpressionSyntax)binaryExpression.Left,
+                    semanticModel,
+                    cancellationToken);
+            }
+            else if (binaryExpression.Kind() == SyntaxKind.EqualsExpression &&
+                     binaryExpression.Right.Kind() == SyntaxKind.NumericLiteralExpression)
+            {
+                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
+                    binaryExpression.Left,
+                    (LiteralExpressionSyntax)binaryExpression.Right,
+                    semanticModel,
+                    cancellationToken);
+            }
+            else if (binaryExpression.Kind() == SyntaxKind.EqualsExpression &&
+                     binaryExpression.Left.Kind() == SyntaxKind.NumericLiteralExpression)
+            {
+                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
+                    binaryExpression.Right,
+                    (LiteralExpressionSyntax)binaryExpression.Left,
+                    semanticModel,
+                    cancellationToken);
+            }
+
+            return canSimplify;
+        }
+
+        private bool CanSimplifyToLengthEqualsZeroExpression(
+            ExpressionSyntax variableExpression,
+            LiteralExpressionSyntax numericLiteralExpression,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var numericValue = semanticModel.GetConstantValue(numericLiteralExpression, cancellationToken);
+            if (numericValue.HasValue && numericValue.Value is int && (int)numericValue.Value == 0)
+            {
+                var symbol = semanticModel.GetSymbolInfo(variableExpression, cancellationToken).Symbol;
+
+                if (symbol != null && (symbol.Name == "Length" || symbol.Name == "LongLength"))
```

Comment:
```nameof(Array.Length)```

Reply:
Thanks @CyrusNajmabad , I tried to do the same with "LongLength" but error message 'Array' does not contain a definition for 'LongLength'.  This project targets netstandard1.3 and Array.LongLength isn't available till netstandard2.0.  I'll use a constant instead.